WcaOnRails/app/helpers/competitions_helper.rb

```diff
@@ -16,16 +16,72 @@ def competition_message_for_user(competition, user)
     messages.join(' ')
   end
 
-  def announced_class(days_announced)
-    level = [Competition::ANNOUNCED_DAYS_WARNING, Competition::ANNOUNCED_DAYS_DANGER].select { |d| days_announced > d }.count
-    ["alert-danger", "alert-orange", "alert-green"][level]
+  private def days_before_competition(date, competition)
+    date ? (competition.start_date - date.to_date).to_i : nil
   end
 
-  def report_and_results_class(days)
+  private def days_after_competition(date, competition)
+    date ? (date.to_date - competition.end_date).to_i : nil
+  end
+
+  def announced_content(competition)
+    days_announced = days_before_competition(competition.announced_at, competition)
+    days_announced ? "#{pluralize(days_announced, "day")} before" : ""
+  end
+
+  def announced_class(competition)
+    days_announced = days_before_competition(competition.announced_at, competition)
+    if days_announced
+      level = [Competition::ANNOUNCED_DAYS_WARNING, Competition::ANNOUNCED_DAYS_DANGER].select { |d| days_announced > d }.count
+      ["alert-danger", "alert-orange", "alert-green"][level]
+    else
+      ""
+    end
+  end
+
+  private def report_and_results_days_to_class(days)
     level = [Competition::REPORT_AND_RESULTS_DAYS_OK, Competition::REPORT_AND_RESULTS_DAYS_WARNING, Competition::REPORT_AND_RESULTS_DAYS_DANGER].select { |d| days > d }.count
     ["alert-green", "alert-success", "alert-orange", "alert-danger"][level]
   end
 
+  def report_content(competition)
+    days_report = days_after_competition(competition.delegate_report.posted_at, competition)
+    if days_report
+      submitted_by_competition_delegate = competition.delegates.include?(competition.delegate_report.posted_by_user)
+      submitted_by_competition_delegate ? "#{pluralize(days_report, "day")} after" : "submitted by else"
```

Comment:
I agree the "submitted by else" sounds weird, that's just what Ilkyoo told me to do. I've change it to "submitted by other".